components/eamxx/src/share/field/field_manager.hpp

```diff
@@ -33,16 +35,17 @@ class FieldManager {
 public:
 
   // Public types
-  using header_type      = typename Field::header_type;
-  using identifier_type  = typename Field::identifier_type;
-  using ci_string        = typename identifier_type::ci_string;
-  using repo_type        = std::map<ci_string,std::shared_ptr<Field>>;
-  using group_info_type  = FieldGroupInfo;
-  using group_info_map   = std::map<ci_string,std::shared_ptr<group_info_type>>;
-  using grid_ptr_type    = std::shared_ptr<const AbstractGrid>;
+  using header_type         = typename Field::header_type;
+  using identifier_type     = typename Field::identifier_type;
+  using ci_string           = typename identifier_type::ci_string;
+  using repo_type           = std::map<ci_string,std::map<ci_string,std::shared_ptr<Field>>>;
+  using grids_mgr_type      = std::shared_ptr<const GridsManager>;
+  using group_info_type     = FieldGroupInfo;
+  using group_info_map      = std::map<ci_string, std::shared_ptr<group_info_type>>;
 
   // Constructor(s)
-  explicit FieldManager (const grid_ptr_type& grid);
+  explicit FieldManager (const std::shared_ptr<const AbstractGrid>& grid);
+  explicit FieldManager (const std::shared_ptr<const GridsManager>& grid);
```

Comment:
Minor: either always use grids_mgr_type, or remove the typedef and just use the spelled out name (like you did for grid). I favor the latter, btw.